bundles/org.openhab.binding.fritzboxtr064/src/main/java/org/openhab/binding/fritzboxtr064/internal/util/SCPDUtil.java

```diff
@@ -53,9 +57,8 @@
 
     private SCPDRootType scpdRoot;
     private final List<SCPDDeviceType> scpdDevicesList = new ArrayList<>();
-    private final Map<String, SCPDScpdType> serviceMap = new HashMap<>();
+    private final Map<String, @Nullable SCPDScpdType> serviceMap = new HashMap<>();
```

Comment:
Same here. It's just the annotations that think that `get` on a `Map` always results in a non-null value. This is simply not true. Adding `@Nullable` means that the value can be null evben if the key was found. That is not the case. `null` is returned if the key is not present.

Reply:
I know, but this is imho the closest solution we have and it is used in many places in our code base already. The main effect here is that get is not expected to be non-null anymore and that's what is correct.